ext/standard/tests/math/number_format_basiclong_64bit.phpt

```diff
@@ -179,8 +179,8 @@ foreach ($numbers as $number) {
 --- testing: float(9.223372036854776E+18)
 ... with precision 5: string(31) "9,223,372,036,854,775,808.00000"
 ... with precision 0: string(25) "9,223,372,036,854,775,808"
-... with precision -1: string(25) "9,223,372,036,854,775,808"
-... with precision -5: string(25) "9,223,372,036,854,800,384"
+... with precision -1: string(25) "9,223,372,036,854,775,800"
```

Comment:
```suggestion
... with precision -1: string(25) "9,223,372,036,854,775,810"
```

to be consistent with https://3v4l.org/ZGU7I

Reply:
mh I could try to use `_php_math_number_format_long` in case the rounded number fits long and decimal places are <= 0

Reply:
because of https://3v4l.org/SoS6N rounding it should be possible to check if the 1st rounded digit is `5-9`

Reply:
The precision is too fine to make much sense in testing the rounding results for this case.

Reply:
I misunderstood. Admittedly, it works a little differently than in 8.3. However, if the documentation is correct, the current behavior is more broken. We need to investigate a little more to see which is the correct behavior.

Reply:
@mvorisek The number here is > 64bit and therefor we have to stick to floating point. The only thing that could be done is manually post rounding based on the rendered digits. What do you think @SakiTakamachi ?

Reply:
Personally, I think it's too fine a precision to do rounding...
However, we need consistency in the specifications, so let's think a little about how we can make it more persuasive.

Reply:
@marc-mabe 
I thought about this for a bit, but if we want accurate results we should use BCMath. It makes no sense to round to such fine precision. It is not possible using normal methods.
https://3v4l.org/N6r4B

If the precision is so fine that rounding is not possible, I think your implementation is fine.

Reply:
this is for double only, for int it is ok - https://3v4l.org/5HqRt, so it seems ok for now

Reply:
Yes, but it exceeds the "meaningful precision" of double.

Reply:
@mvorisek formatting big floating point numbers still within the range of int are using the integer variant already (which internally uses unsigned long to be able to still round up). See https://github.com/php/php-src/pull/12333